packages/tabs/src/index.tsx

```diff
@@ -455,6 +455,8 @@ export const Tab = forwardRefWithAs<
     context: TabsDescendantsContext,
     disabled: !!disabled,
   });
+  const htmlType =
+    Comp === "button" && props.type == null ? "button" : props.type;
```

Comment:
The reason this was removed to begin with is because this check isn't reliable. `Comp` might be a styled component, or a wrapper component of some kind, in which case we'd lose that default setting. Of course if we pass the attribute along to a non-button element then we have invalid HTML. May still be worth it to have either way.

Reply:
Maybe we could keep the check, since it would cover at least some cases, and add some information in the docs about the edge cases?